internal/executor/instance/persistentworker/isolation/parallels/vm.go

```diff
@@ -0,0 +1,119 @@
+package parallels
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"strings"
+)
+
+var ErrVMFailed = errors.New("Parallels VM operation failed")
+
+type VM struct {
+	name string
+}
+
+type NetworkAdapterInfo struct {
+	MAC string
+}
+
+type HardwareInfo struct {
+	Net0 NetworkAdapterInfo
+}
+
+type VirtualMachineInfo struct {
+	ID       string
+	Name     string
+	Hardware HardwareInfo
+}
+
+func NewVMClonedFrom(ctx context.Context, vmNameFrom string) (*VM, error) {
+	vm := &VM{
+		name: "cirrus-" + uuid.New().String(),
+	}
+
+	_, stderr, err := Prlctl(ctx, "clone", vmNameFrom, "--name", vm.name)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to clone VM %q: %q", ErrVMFailed, vm.name, firstLine(stderr))
+	}
+
+	// Ensure that the VM is isolated[1] from the host (e.g. shared folders, clipboard, etc.)
+	// nolint:lll // https://github.com/walle/lll/issues/12
+	// [1]: https://download.parallels.com/desktop/v14/docs/en_US/Parallels%20Desktop%20Pro%20Edition%20Command-Line%20Reference/43645.htm
+	_, stderr, err = Prlctl(ctx, "set", vmNameFrom, "--isolate-vm", "on")
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to isolate VM %q: %q", ErrVMFailed, vm.name, firstLine(stderr))
+	}
+
+	_, stderr, err = Prlctl(ctx, "start", vm.name)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to start VM %q: %q", ErrVMFailed, vm.name, firstLine(stderr))
+	}
+
+	return vm, nil
+}
+
+func (vm *VM) Close() error {
+	ctx := context.Background()
+
+	_, stderr, err := Prlctl(ctx, "stop", vm.name, "--kill")
+	if err != nil {
+		return fmt.Errorf("%w: failed to stop VM %q: %q", ErrVMFailed, vm.name, firstLine(stderr))
+	}
+
+	_, stderr, err = Prlctl(ctx, "delete", vm.name)
+	if err != nil {
+		return fmt.Errorf("%w: failed to delete VM %q: %q", ErrVMFailed, vm.name, firstLine(stderr))
+	}
+
+	return nil
+}
+
+func (vm *VM) retrieveInfo(ctx context.Context) (*VirtualMachineInfo, error) {
+	stdout, stderr, err := Prlctl(ctx, "list", "--info", "--json")
```

Comment:
You can pass `vm.name` here:

```suggestion
	stdout, stderr, err := Prlctl(ctx, "list", "--info", "--json", vm.name)
```

Just a small optimization